torch/csrc/jit/codegen/cuda/shape_inference.cpp

```diff
@@ -154,6 +168,27 @@ class NaiveShapeTypePropagator {
   }
 
  protected:
+  TensorTypePtr unary_reduce_type(
+      TensorTypePtr const& op,
+      const std::vector<int64_t>& dims,
+      bool keepdim) {
+    TORCH_CHECK(
+        op->scalarType().has_value() && op->device().has_value() &&
+            op->sizes().isComplete(),
+        "requires complete shape on input");
+    std::vector<int64_t> output_size;
+    std::vector<int64_t> input_size = *op->sizes().concrete_sizes();
+    for (size_t i = 0; i < input_size.size(); i++) {
+      if (std::find(dims.begin(), dims.end(), i) == dims.end()) {
+        output_size.emplace_back(input_size[i]);
+      } else if (keepdim) {
+        output_size.emplace_back(1);
```

Comment:
what is the meaning of 1 here?

Reply:
for `x = torch.randn(10, 10).sum(axis=0, keep_dim=True)`, `x.size()` would give us `[1, 10]`; if we do `keep_dim=False`, `x.size()` gives `[10]`.

So when keepdim is true, we want to keep the dimension by setting its size to 1.